cmd/chihaya/main.go

```diff
@@ -65,9 +65,17 @@ func (r *Run) Start(ps storage.PeerStore) error {
 	if err != nil {
 		return errors.New("failed to validate hook config: " + err.Error())
 	}
+	preHookNames := make([]string, 0)
+	for _, hookCfg := range cfg.PreHooks {
+		preHookNames = append(preHookNames, hookCfg.Name)
+	}
```

Comment:
`prehooks` and `posthooks` are of type `[]middleware.Hook`, which does not have the name field like `hookConfig`. It will be more like:

```go
log.WithFields(log.Fields{
  "prehooks": cfg.PreHooks.Names(),
  "posthooks": cfg.PostHooks.Names(),
}).Info("starting middleware")
```
The `Names()` method would be for type `[]hookConfig`.

OK?